src/main/java/de/dennisguse/opentracks/util/FileUtils.java

```diff
@@ -289,6 +289,30 @@ public static File getPhotoFileIfExists(Context context, Track.Id trackId, Uri u
         return file;
     }
 
+    /**
+     * Builds interval photo file object for fileNameUri Uri.
+     *
+     * @param context     the Context.
+     * @param trackId     the id of the Track.
+     * @param fileNameUri the file name uri.
+     * @return            file object or null.
+     */
+    public static File buildInternalPhotoFile(Context context, Track.Id trackId, Uri fileNameUri) {
```

Comment:
fileNameUri should be @NonNull and the null check should be done by the caller (if needed).